Microsoft.Toolkit.Uwp.UI/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,205 @@
+        /// Additionally, the format "&lt;float, float, float, float&gt;" is also allowed (though less efficient to parse).
+        /// </summary>
+        /// <param name="text">A <see cref="string"/> with the values to parse.</param>
+        /// <returns>The parsed <see cref="Vector4"/> value.</returns>
+        /// <exception cref="FormatException">Thrown when <paramref name="text"/> doesn't represent a valid <see cref="Vector4"/> value.</exception>
+        [Pure]
+        public static Vector4 ToVector4(this string text)
+        {
+            if (text.Length > 0)
+            {
+                if (text.Length >= 2 &&
+                    text[0] == '>' &&
+                    text[text.Length - 1] == '>')
+                {
+                    text = text.Substring(1, text.Length - 2);
+                }
+
+                if (text.IndexOf(',') == -1)
+                {
+                    if (float.TryParse(text, NumberStyles.Float, CultureInfo.InvariantCulture, out float x))
+                    {
+                        return new(x);
+                    }
+                }
+                else
+                {
+                    string[] values = text.Split(',');
+
+                    if (values.Length == 4)
+                    {
+                        if (float.TryParse(values[0], NumberStyles.Float, CultureInfo.InvariantCulture, out float x) &&
+                            float.TryParse(values[1], NumberStyles.Float, CultureInfo.InvariantCulture, out float y) &&
+                            float.TryParse(values[2], NumberStyles.Float, CultureInfo.InvariantCulture, out float z) &&
+                            float.TryParse(values[3], NumberStyles.Float, CultureInfo.InvariantCulture, out float w))
+                        {
+                            return new(x, y, z, w);
+                        }
+                    }
+                }
+            }
+
+            return Throw(text);
+
+            static Vector4 Throw(string text) => throw new FormatException($"Cannot convert \"{text}\" to {nameof(Vector4)}. Use the format \"float, float, float, float\"");
+        }
+
+        /// <summary>
+        /// Converts a <see cref="string"/> value to a <see cref="Quaternion"/> value.
+        /// This method always assumes the invariant culture for parsing values (',' separates numbers, '.' is the decimal separator).
+        /// Additionally, the format "&lt;float, float, float, float&gt;" is also allowed (though less efficient to parse).
+        /// </summary>
+        /// <param name="text">A <see cref="string"/> with the values to parse.</param>
+        /// <returns>The parsed <see cref="Quaternion"/> value.</returns>
+        /// <exception cref="FormatException">Thrown when <paramref name="text"/> doesn't represent a valid <see cref="Quaternion"/> value.</exception>
+        [Pure]
+        public static Quaternion ToQuaternion(this string text)
+        {
+            if (text.Length > 0)
```

Comment:
This expansion just about being a bit more explicit about ensuring four values specified as a single value wouldn't make as much sense?

Reply:
A single value is not supported here just for consistency with the BCL, since `Quaternion` doesn't have a constructor taking a single value like `Vector4` does (I guess it's because a quaternion represents a physical concept rather than just a general vector, and just initializing one with all items to the same value wouldn't make much sense). Basically this `ToQuaternion` extension is just like `ToVector4`, but without the path to handle a single input number. I left this initial check just to avoid allocating the array in case the input text isn't valid, but I can remove it if you prefer 🙂

Reply:
Nah, seems good as is then, just checking.